daemon/networkdriver/bridge/driver.go

```diff
@@ -230,17 +233,25 @@ func InitDriver(job *engine.Job) error {
 		}
 	}
 
+	if hairpinMode {
+		// Enable loopback adresses routing
+		sysPath := filepath.Join("/proc/sys/net/ipv4/conf", bridgeIface, "route_localnet")
+		if err := ioutil.WriteFile(sysPath, []byte{'1', '\n'}, 0644); err != nil {
+			logrus.Warnf("WARNING: unable to enable local routing for hairpin mode: %s\n", err)
```

Comment:
I think the `WARNING:` at the beginning of the message is superfluous and it can be removed, logrus already adds a `WARNING` to the log.